index.js

```diff
@@ -290,7 +290,12 @@ function parse (args, opts) {
   // e.g., --foo apple banana cat becomes ["apple", "banana", "cat"]
   function eatArray (i, key, args) {
     for (var ii = i + 1; ii < args.length; ii++) {
-      if (/^-/.test(args[ii])) break
+      if (/^-/.test(args[ii])) {
+        if (args[ii - 1].replace('-', '') === key) {
```

Comment:
@bcoe Won't this also return true in any other iteration? Since `i` gets assigned the value of `ii` at the end of each iteration, and `ii` gets incremented at the start of every iteration, doesn't that mean that `ii === i + 1` will always return true?

I still have passing tests, but I think that's because `yargs-parser` seems to only run arrays through `eatArray()` if they're short grouped and alone as the first option, like this:
```javascript
const yargs = require('yargs');

const argv = yargs
  .array('a')
  .array('b')
  .argv;
```
```
myModule -a -b
```
^ `eatArray()` is only run against `a`. If you call via `myModule -b -a` instead, it's run only against `b`. Is this intentional?